client/my-sites/theme/bundle-settings.tsx

```diff
@@ -0,0 +1,32 @@
+import { translate } from 'i18n-calypso';
+
+interface BundleSettings {
+	[ key: string ]: {
+		name: string;
+		BadgeIcon: React.ComponentType;
+		designPickerBadgeTooltip: string;
+		bannerUpsellDescription: string;
+	};
+}
+
+const bundleSettings: BundleSettings = {
```

Comment:
Have you considered wrapping this in a hook? I am asking because the current direct lookups in `bundleSettings[ type ]` feel a bit awkward _and_ because the use of `translate` directly may miss contextual information that's needed to get the right locale etc. If we're in a hook, we can switch to `const translate = useTranslate();` within the hook.

Reply:
Gotcha! I didn't consider this scenario where the `translate` could change.
Changed it here: https://github.com/Automattic/wp-calypso/pull/84483/commits/76d49f2aa201ceb84ff9317478fac4f334bc6997
And then simplified a bit more here: https://github.com/Automattic/wp-calypso/pull/84483/commits/ab8af367e962efadaac36b1d6ce7095337a801e6